activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -356,7 +356,11 @@ def find_with_associations
 
     def construct_join_dependency(joins = [])
       including = eager_load_values + includes_values
-      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins)
+      finder_methods_hash = {
+        eager_load: eager_load_values,
+        includes: includes_values
+      }
+      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins, finder_methods_hash)
```

Comment:
I wonder if we should stop passing the `including` here, as we will have access to the `eager_load_values` and `includes_values` now.